### PR TITLE
Make zoom panning slippy, like in Google Maps

### PR DESCRIPTION
**Pull request by jacobp100**
https://ag-grid.atlassian.net/browse/AG-10941

### CODE REVIEW

**Review by lsjroberts — APPROVED** [review]
lgtm, thanks!

File: packages/ag-charts-enterprise/src/features/zoom/zoomPanner.ts [hunk 1]
```diff
@@ -1,19 +1,136 @@
...
+        this.zoomCoordsHistoryIndex = 0;
+        this.coordsHistory.length = 0;
+
+        if (deltaT > 0 && this.deceleration < 1) {
+            const xVelocity = deltaX / deltaT;
+            const yVelocity = deltaY / deltaT;
+            const velocity = Math.hypot(xVelocity, yVelocity);
+            const angle = Math.atan2(yVelocity, xVelocity);
+            const t0 = performance.now();
+            this.inertiaHandle = requestAnimationFrame((t) => {
```
Comment (alantreadway) [inline packages/ag-charts-enterprise/src/features/zoom/zoomPanner.ts:97]: I'm not sure if we can use `requestAnimationFrame()` directly (here and below) - if chart rendering is slower than 60 FPS, I'm concerned that the updates this generates will not play well with an in-progress Chart.performUpdate() cycle.

Could we adjust the solution to use `AnimationManager` to execute the animation steps, as it already handles coordination with `Chart.performUpdate()`?

Comment (alantreadway) [reply packages/ag-charts-enterprise/src/features/zoom/zoomPanner.ts:97]: Discussed offline - as this uses `ZoomManager` to percolate state changes, we are already taking this concern indirectly into account.

**Review by alantreadway — APPROVED** [review]
LGTM

**Pull request merged**